src/attr/__init__.pyi

```diff
@@ -49,19 +53,26 @@ def fields(cls: type) -> Tuple[Attribute, ...]: ...
 def validate(inst: Any) -> None: ...
 
 # we use Any instead of _CountingAttr so that e.g. `make_class('Foo', [attr.ib()])` is valid
-def make_class(name, attrs: Union[List[Any], Dict[str, Any]], bases: Tuple[type, ...] = ..., **attributes_arguments) -> type: ...
+def make_class(name, attrs: Union[List[str], Dict[str, Any]], bases: Tuple[type, ...] = ..., **attributes_arguments) -> type: ...
 
 # _funcs --
 
-@overload
-def asdict(inst: Any, *, recurse: bool = ..., filter: Optional[_FilterType] = ..., dict_factory: Type[_M], retain_collection_types: bool = ...) -> _M: ...
-@overload
-def asdict(inst: Any, *, recurse: bool = ..., filter: Optional[_FilterType] = ..., retain_collection_types: bool = ...) -> Dict[str, Any]: ...
+# FIXME: asdict/astuple do not honor their factory args.  waiting on one of these:
+# https://github.com/python/mypy/issues/4236
+# https://github.com/python/typing/issues/253
+def asdict(inst: Any, *, recurse: bool = ..., filter: Optional[_FilterType] = ..., dict_factory: Type[Mapping] = ..., retain_collection_types: bool = ...) -> Dict[str, Any]: ...
```

Comment:
You should remove the `*` since these are not kwarg only arguments.

Reply:
ah, yes, that's left over from my struggle with overloads, before I gave up on them.